tasks/species_typing/task_tbprofiler_output_parsing.wdl

```diff
@@ -285,117 +552,130 @@ task tbprofiler_output_parsing {
 
         # reported mutation by tb-profiler, all confering resistance by WHO criteria
         for dr_variant in results_json["dr_variants"]: 
+          genes_reported.append(dr_variant["gene"])
           if "annotation" in dr_variant:
-            try:  # sometimes annotation is an empty list
-              if dr_variant["annotation"][0]["who_confidence"] == "":
-                confidence = "No WHO annotation"
-              else:
-                confidence = dr_variant["annotation"][0]["who_confidence"]
-            except:
-              confidence = "No WHO annotation"
+            # empty dictionary in the case where multiple drugs are identified
+            drugs_to_row = {}
+            # case: annotation field is present but it's an empty list, expert rule is applied directly 
+            if len(dr_variant["annotation"]) == 0:
+              row = variant_to_row(dr_variant)
+              row["confidence"] = "No WHO annotation"
+              row["looker_interpretation"] = apply_expert_rules(dr_variant["nucleotide_change"], dr_variant["protein_change"], dr_variant["gene"], dr_variant["type"], "looker")
+              row["mdl_interpretation"] = apply_expert_rules(dr_variant["nucleotide_change"], dr_variant["protein_change"], dr_variant["gene"], dr_variant["type"], "MDL")
+              row["rationale"] = "Expert rule applied"
+
+              # iterate through any gene-associated drugs
```

Comment:
considering moving the comments before or after line, to make things a little more readable